### PR TITLE
Odometry fix

### LINKED ISSUE #27: Inconsistencies in odometry update
This Notifier is running at 200Hz to call odometry update.
Periodic is running at 50Hz and includes an update method.
Only one of these should be updating the odometry.

If you follow the notifier call stack, it goes into the swerve module to get position. This method returns the values stored in the drivePosition and turnPosition variables. These variables are updated in periodic() which is running at 50Hz.

Option 1: remove the Notifier.
Option 2: make sure the Notifier is receiving fresh data at 200Hz. This includes making sure that encoders are reporting updates on CAN at a frequency greater than 200Hz.

https://github.com/koibots8230/RumbleBot2024/blob/7d6cb0e2766013795415bc9384092a4b5591cb52/src/main/java/frc/robot/subsystems/swerve/Swerve.java#L77

### PR DESCRIPTION
**Pull request by JakeStew81**
# Pull request for Odometry Fix

## Included Features:

- Odometry actually runs at 200hz

## Checklist:

- [x] The code builds
- [x] The code has been self or peer-reveiwed
- [ ] The code works in simulation
- [ ] The code has been tested on the robot & works as intended

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/swerve/SwerveModule.java [hunk 1]
```diff
@@ -106,6 +111,12 @@ public SwerveModule(int turnID, int driveID, Rotation2d angleOffset) {
     driveMotor.enableVoltageCompensation(RobotConstants.NOMINAL_VOLTAGE.in(Volts));
     driveMotor.setCANTimeout((int) RobotConstants.CAN_TIMEOUT.in(Milliseconds));
 
+    driveMotor.setPeriodicFramePeriod(PeriodicFrame.kStatus2, 3);
+    driveMotor.setPeriodicFramePeriod(PeriodicFrame.kStatus3, 32767);
```
Comment (eskiles) [inline src/main/java/frc/robot/subsystems/swerve/SwerveModule.java:115]: The frame period is set to 32.767 seconds??

File: src/main/java/frc/robot/subsystems/swerve/Swerve.java [hunk 2]
```diff
@@ -93,8 +93,22 @@ public Swerve(boolean isReal) {
...
-                    Timer.getFPGATimestamp(), gyro.getRotation2d(), getModulePositions());
+                try {
+                  modules[0].periodic();
+                  modules[1].periodic();
+                  modules[2].periodic();
+                  modules[3].periodic();
+
+                  odometry.updateWithTime(
+                      Timer.getFPGATimestamp(),
+                      (DriverStation.getAlliance().get() == DriverStation.Alliance.Blue)
```
Comment (eskiles) [inline src/main/java/frc/robot/subsystems/swerve/Swerve.java:104]: Why does the robot pose depend on the alliance? The field is fixed and the robot pose should be the same whether it is on red or blue alliance.

File: src/main/java/frc/robot/subsystems/swerve/SwerveModule.java (hunk 1 above)
Comment (JakeStew81) [reply src/main/java/frc/robot/subsystems/swerve/SwerveModule.java:115]: It's just to reduce CAN utilization. That periodic frame has data we don't use, and 32767ms is the highest you can set it to (REV doesn't let you disable it completely to my knowledge).

File: src/main/java/frc/robot/subsystems/swerve/Swerve.java (hunk 2 above)
Comment (JakeStew81) [reply src/main/java/frc/robot/subsystems/swerve/Swerve.java:104]: The field is fixed but the zero angle of the gyro isn't

**Pull request merged**